CampusTour/ViewControllers/PopupViewController.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  PopupViewController.swift
+//  CampusTour
+//
+//  Created by Ji Hwan Seung on 4/5/18.
+//  Copyright © 2018 cuappdev. All rights reserved.
+//
+
+import UIKit
+
+protocol PopupFilterProtocol {
+    func updateFilterBar(_ status: FilterBarCurrentStatus) -> ()
+}
+
+class PopupViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
+    
+    var data: PopupData = PopupData(filterBarStatus: FilterBarCurrentStatus(), filterMode: Filter.general, filterBarLocationCenterX: 0)
+    let reuseID = "reuseID"
+    let tableView = UITableView()
+    var triangleView: TriangleView!
+    private let triangleViewLength = CGFloat(10)
+    private var currentCheckedCellIndex: IndexPath?
+    var delegate: PopupFilterProtocol!
+    
+    override func viewDidLoad() {
+        view.backgroundColor = .clear
+        
+        triangleView = TriangleView(frame: CGRect(x: 0, y: 0, width: triangleViewLength, height: triangleViewLength))
+        triangleView.backgroundColor = .clear
+        
+        tableView.layer.cornerRadius = 5.0
+        tableView.clipsToBounds = true
+        
+        view.addSubview(tableView)
+        view.addSubview(triangleView)
+        
+        triangleView.snp.makeConstraints { (make) in
+            make.centerX.equalTo(data.filterBarLocationCenterX)
+            make.top.equalToSuperview()
+            make.height.equalTo(triangleViewLength)
+            make.width.equalTo(triangleViewLength)
+        }
+        tableView.snp.makeConstraints { (make) in
+            make.top.equalTo(triangleView.snp.bottom)
+            make.leading.equalToSuperview()
+            make.trailing.equalToSuperview()
+            make.bottom.equalToSuperview()
+        }
+        tableView.register(FilterTableViewCell.self, forCellReuseIdentifier: reuseID)
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.backgroundColor = .white
+        tableView.bounces = false
+    }
+    
+    override func updateViewConstraints() {
+        triangleView.snp.updateConstraints { (make) in
+            make.centerX.equalTo(data.filterBarLocationCenterX)
+            make.top.equalToSuperview()
+            make.height.equalTo(triangleViewLength)
+            make.width.equalTo(triangleViewLength)
+        }
+        super.updateViewConstraints()
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
```

Comment:
I feel like using a tableview is a bit overkill here. a stack inside of a scroll view would work will less code imo

Reply:
Oo yeah could’ve done that, but changing it will be too big of a change rn I think